File: test/clj/truffler/environment_test.clj
(ns truffler.environment-test
    (:require [clojure.test :refer :all])
    (:import [truffler.simple.env Environment]
             [truffler.simple.form SymbolForm]))

(deftest put-and-get
  (let [env (Environment.)]
    (.putValue env (SymbolForm. "foo") "bar")
    (is (= "bar"
           (.getValue env (SymbolForm. "foo"))))))

File: simple/test/mumbler/simple/env/EnvironmentTest.java
package mumbler.simple.env;

import static org.junit.Assert.*;

import org.junit.Before;
import org.junit.Test;

public class EnvironmentTest {
    
    private Environment env;
    
    @Before
    public void setUp() {
        this.env = new Environment();
    }
    
    @Test
    public void testGetValue() {
        this.env.putValue("foo", new Long(1));
        assertEquals(1L, this.env.getValue("foo"));
    }

}
